android-core/src/main/java/com/mparticle/internal/DeviceAttributes.kt

```diff
@@ -0,0 +1,343 @@
+package com.mparticle.internal
+
+import android.app.Application
+import android.content.Context
+import android.content.pm.PackageManager
+import android.os.Build
+import android.telephony.TelephonyManager
+import com.mparticle.MParticle
+import com.mparticle.internal.Constants.MessageKey
+import com.mparticle.internal.Constants.PrefKeys
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.Locale
+import java.util.TimeZone
+
+class DeviceAttributes
+/**
+ * package-private
+ */ internal constructor(private val operatingSystem: MParticle.OperatingSystem?) {
+    private var deviceInfo: JSONObject? = null
+    private var appInfo: JSONObject? = null
+    private var firstCollection = true
+
+    private val sideloadedKitsCount: Int
+        get() {
+            try {
+                val kits = MParticle.getInstance()?.Internal()?.kitManager?.supportedKits
+                var count = 0
+                if (kits != null) {
```

Comment:
More of a style question, but is it more idiomatic to use `.let {}` instead of `if != null`? Are there any downsides to using `.let`? 

In Swift at least, it's considered non-idiomatic to do null checks this way anymore vs `if let variable = variable` which I think is similar in principle to the `.let{}` in kotlin.

Reply:
We can use let{} here, but since we're using a for loop with the kits instance, we should use an if condition instead of let.